backend/src/main/kotlin/com/tngtech/apicenter/backend/domain/entity/Specification.kt

```diff
@@ -1,11 +1,29 @@
 package com.tngtech.apicenter.backend.domain.entity
 
+import java.util.regex.Pattern
+
+enum class ReleaseType {
+    RELEASE, PRERELEASE, SNAPSHOT
+}
+
+fun inferReleaseType(version: String): ReleaseType {
+    return if (version.endsWith("-SNAPSHOT")) {
```

Comment:
First, I'd like those patterns in a constant (you should be able to do `-(BETA|RC)` for the `PRERELEASE`.
Secondly, I think this should be a `fromVersionString` method in the `enum class` itself.

Reply:
To the first point (after belatedly actually reading the semver spec...), it seems we ought to be a bit more liberal in the pre-release acceptance criteria? https://semver.org/#spec-item-9 (Or would you rather only have these extensions?)

Reply:
I can live with both. On the one hand, supporting the full Semver spec would be nice, but on the other hand, one should not have different versions of one version...

So whatever is easier, I'd say